custom_dc/README.md

```diff
@@ -0,0 +1,227 @@
+html and css customization. These customization are in the following folders:
+
+- html files: [location](../server/templates/custom_dc/custom/)
+- css and image files: [location](../static/custom_dc/custom/)
+
+Update these files for UI customization.
+
+## Local Development
+
+Custom Data Commons site via [localhost](http://localhost:8080) by following the
+instructions below:
+
+### Build docker image locally
+
+```bash
+docker build --tag datacommons-website-compose:latest \
+-f build/web_compose/Dockerfile \
+-t website-compose .
+```
+
+### Run custom Data Commons locally with SQLITE database
+
+In the root of this repository, run:
+
+```bash
+docker run -it \
+--env-file $PWD/custom_dc/sqlite_env.list \
+-p 8080:8080 \
+-v $PWD/custom_dc/sample:/sqlite \
+-v $PWD/server/templates/custom_dc:/workspace/server/templates/custom_dc \
+datacommons-website-compose:latest
+```
+
+## Run in Cloud
+
+Custom Data Commons can be ran on the cloud as a production service. The SQLite
+approach can still be used. Note you need to copy the data folders into the
+cloud disk and mount it to the docker container then specify the enviornment
+variables based on the requirements of the cloud providers.
+
+We have also provided a specific deployment setup on Google Cloud Run that is
+based on Cloud SQL.
+
+### Setup Cloud SQL
+
+Create a Cloud SQL instance from the [Cloud
+Console](https://console.cloud.google.com/sql/instances). Set instance ID as
+`dc-graph`, choose the type as "MySQL" and create a database `datacommons`. You
+will need to set a user and password. Record the instance connection name in the
+form of "<project>:<region>:dc-graph"
+
+### Setup Google Cloud Storage
+
+Google Cloud Storage is used to hold the data CSV files. From the [Cloud
+Console](https://console.cloud.google.com/storage/browser), create a new bucket
+or pick an existing bucket and upload the data CSV files there. It's recommended
+to create intermediate folders for the files for easier management.
+
+### Upload Data Files
+
+Upload the data CSV files into GCS and record the folder path as
+`gs://<bucket-name>/.../`.
+
+### Test Locally
+
+Authentical Google Cloud:
+
+```bash
+gcloud auth application-default login
+```
+
+This should generate a credential json file in
+`$HOME/.config/gcloud/application_default_credentials.json`. This is used in the
+command below for authentication in the docker container.
+
+In the root of this repository, run:
+
+```bash
+docker run -it \
+--env-file $PWD/custom_dc/cloudsql_env.list \
+-p 8080:8080 \
+-e GOOGLE_APPLICATION_CREDENTIALS=/gcp/creds.json \
+-v $HOME/.config/gcloud/application_default_credentials.json:/gcp/creds.json:ro \
+-v $PWD/server/templates/custom_dc:/workspace/server/templates/custom_dc \
+datacommons-website-compose:latest
+```
+
+[Note] you can change the docker image to use your custom built docker image.
+
+### Deploy to Cloud Run
+
+Build docker image and push it to Google Container Registry
```

Comment:
I believe GCR is deprecated now -- I ended up using Artifact Registry. Either way, should mention the prerequisite of setting up the registry first

Reply:
Changed to use artifact registry, this is a very good move.